otelcol/flags_test.go

```diff
@@ -54,11 +54,28 @@ func TestSetFlag(t *testing.T) {
 			args:        []string{"--set=key:name"},
 			expectedErr: `invalid value "key:name" for flag -set: missing equal sign`,
 		},
+		{
+			name: "feature gates",
+			args: []string{"--feature-gates=alpha,-beta"},
+		},
+		{
+			name:        "feature gates and strict",
+			args:        []string{"--feature-gates=alpha,-beta", "--feature-gates-strict=true"},
+			expectedErr: `invalid boolean value "true" for -feature-gates-strict: gate "beta" is in beta and must be explicitly enabled`,
+		},
+		{
+			name:        "strict flag first and feature gates",
+			args:        []string{"--feature-gates-strict=true", "--feature-gates=alpha,-beta"},
+			expectedErr: `invalid value "alpha,-beta" for flag -feature-gates: gate "beta" is in beta and must be explicitly enabled`,
+		},
```

Comment:
Varying the behavior based on the order of the flags doesn't seem ideal and may confuse users.

Reply:
I could not find a way around this limitation of go flags.